central/sensor/service/pipeline/imageintegrations/pipeline_test.go

```diff
@@ -146,3 +156,121 @@ func Test_matchesECRAuth(t *testing.T) {
+			ImagePullSecretName: "fake-secretname",
+		},
+	}
+
+	t.Run("no integrations sourced when features disabled", func(t *testing.T) {
+		testutils.MustUpdateFeature(t, features.SourcedAutogeneratedIntegrations, false)
+		t.Setenv(env.AutogenerateGlobalPullSecRegistries.EnvVar(), "false")
+
+		assert.False(t, sourcedImageIntegration(iiNoSrc))
+		assert.False(t, sourcedImageIntegration(iiGlobalPull))
+		assert.False(t, sourcedImageIntegration(iiSourcedOther))
+	})
+
+	t.Run("all integrations with source are sourced when main feature enabled", func(t *testing.T) {
+		testutils.MustUpdateFeature(t, features.SourcedAutogeneratedIntegrations, true)
+		t.Setenv(env.AutogenerateGlobalPullSecRegistries.EnvVar(), "false")
+
+		assert.False(t, sourcedImageIntegration(iiNoSrc))
+		assert.True(t, sourcedImageIntegration(iiGlobalPull))
+		assert.True(t, sourcedImageIntegration(iiSourcedOther))
+	})
+
+	t.Run("all integrations with source are sourced when all feature enabled", func(t *testing.T) {
+		testutils.MustUpdateFeature(t, features.SourcedAutogeneratedIntegrations, true)
+		t.Setenv(env.AutogenerateGlobalPullSecRegistries.EnvVar(), "true")
+
+		assert.False(t, sourcedImageIntegration(iiNoSrc))
+		assert.True(t, sourcedImageIntegration(iiGlobalPull))
+		assert.True(t, sourcedImageIntegration(iiSourcedOther))
+	})
+
+	t.Run("only global pull integration sourced when global pull feature enabled", func(t *testing.T) {
+		testutils.MustUpdateFeature(t, features.SourcedAutogeneratedIntegrations, false)
+		t.Setenv(env.AutogenerateGlobalPullSecRegistries.EnvVar(), "true")
+
+		assert.False(t, sourcedImageIntegration(iiNoSrc))
+		assert.True(t, sourcedImageIntegration(iiGlobalPull))
+		assert.False(t, sourcedImageIntegration(iiSourcedOther))
+	})
+}
+
+func TestSetupIntegrationParamsDescription(t *testing.T) {
+	ctx := context.Background()
+	ii := &storage.ImageIntegration{
+		IntegrationConfig: &storage.ImageIntegration_Docker{
+			Docker: &storage.DockerConfig{
+				Endpoint: "registry.invalid/path/to/thing",
+			},
+		},
+	}
+
+	t.Run("do not include path in description", func(t *testing.T) {
+		desc, _, _ := setUpIntegrationParams(ctx, ii.CloneVT(), false)
+		assert.NotContains(t, desc, "/path")
+		assert.Equal(t, "https://registry.invalid", desc)
+	})
+
+	t.Run("include path in description", func(t *testing.T) {
+		desc, _, _ := setUpIntegrationParams(ctx, ii.CloneVT(), true)
+		assert.Equal(t, "https://registry.invalid/path/to/thing", desc)
+	})
+}
+
+func TestReconcileGlobalPullSecret(t *testing.T) {
+	bCtx := context.Background()
+	clusterID := "cluster-id"
+
+	t.Run("no reconcile when features disabled", func(t *testing.T) {
+		testutils.MustUpdateFeature(t, features.SourcedAutogeneratedIntegrations, false)
+		t.Setenv(env.AutogenerateGlobalPullSecRegistries.EnvVar(), "false")
+
+		p := pipelineImpl{}
+		assert.NoError(t, p.Reconcile(bCtx, clusterID, nil))
+	})
+
+	t.Run("reconcile when feature enabled", func(t *testing.T) {
```

Comment:
technically the "feature" is disabled and the "env var" is enabled :D

Reply:
It depends 😉 - Here I was referring to a 'feature' generally as a piece of functionality, not one of the flags in the `features` package. 

> technically the "feature" is disabled and the "env var" is enabled :D

But what if the env var's name is `CoolFeatureDisabled` - then would argue not true ^^. 

The changeable flags in the `features` package and the boolean flags in the `env` package are no different (IMHO) outside of values in the `features` package are accessible via the API.